app/src/screens/ProposalDetail.js

```diff
@@ -43,13 +50,51 @@ function ProposalDetail({ proposal, onBack, requestToken }) {
     executed,
   } = proposal
 
-  const handleWithdraw = useCallback(() => {
-    api.withdrawAllFromProposal(id).toPromise()
-  }, [api])
+  const { stakes, threshold } = getStakesAndThreshold(proposal)
+  const conviction = getCurrentConviction(stakes, blockNumber, alpha)
+  const myStakes = stakes.filter(({ entity }) => entity === connectedAccount)
+  const didIStaked = myStakes.length > 0 && [...myStakes].pop().tokensStaked > 0
 
   const handleExecute = useCallback(() => {
     api.executeProposal(id, true).toPromise()
-  }, [api])
+  }, [api, id])
+
+  const handleStake = useCallback(() => {
+    api.stakeAllToProposal(id).toPromise()
+  }, [api, id])
+
+  const handleWithdraw = useCallback(() => {
+    api.withdrawAllFromProposal(id).toPromise()
+  }, [api, id])
+
+  const mode = useMemo(() => {
+    if (conviction.gte(threshold)) {
+      return 'execute'
+    }
+    if (didIStaked) {
+      return 'update'
+    }
+    return 'support'
+  }, [conviction, threshold, didIStaked])
```

Comment:
Looks like we could use this conditions directly on the `useMemo` below and drop the `mode` ? 

Mainly because we are not using the `mode` variable anywhere else.